### ISSUE TITLE
[BUG] CPU and Disk settings are not applied in clone.

### ISSUE THREAD

**Issue by itek09** [danitso/terraform-provider-proxmox#13]
Hello!

In first place I want to say thank you for your work and your amazing provider, It's really useful and let me automate some process in my proxmox server.

**About the issue:**
I'm creating a module to create VM using your provider, but I detect a few thinks, all of them when I clone from template (configured with 2 cores, 1 socket, 10gb hdd):

If I try to modify the cpu settings like this:
```
  cpu {
    cores = 4
    sockets = 1
  }
```
the created VM will maintain the same CPU configuration than the template (2 cores 1 socket in my template).

The same with the HDD configuration:
```
  disk {
    datastore_id = "shared"
    size         = 20
  }
```
The HDD will continue with 10 gb, like the template.

The same with the memory, do not change from the template.

**To Reproduce**

I will paste an example main.tf code to replicate my code:

```
provider "proxmox" {
  virtual_environment {
    endpoint = "${var.pm_api_url}"
    username = "${var.pm_user}"
    password = "${var.pm_password}"
    insecure = "${var.pm_tls_insecure}"
  }
}

resource "proxmox_virtual_environment_vm" "vm" {
  count      = 1
  name = "clone1"
  node_name = "node1"
  cpu {
    cores = "4"
    sockets = "1"
  }
  clone {
    vm_id = "5" 
#Template with 2 cores, 10gb HDD and 2048 RAM
  }
  disk {
    datastore_id = "shared"
    size         =  20
  }
  memory {
    dedicated = 4096
  }

  initialization {
    datastore_id = "shared"

    ip_config {
      ipv4 {
        address = "${var.pm_network_ip[count.index]}"
        gateway = "${var.pm_network_gateway}"
      }
    }
    user_account {
      keys     = ["${file(var.ssh_public_key)}"]
    }
  }

  network_device {
    bridge  = "${var.pm_network_bridge}"
    model   = "${var.pm_network_model}"
    vlan_id = "${var.pm_vlan}"
  }

}

```
You can modify the code to adapt it to you server.

**Expected behavior**
A new virtual machine with modified ram, cpu and hdd is expected.
A new virtual machine with same ram, cpu and hdd settings than the clone is created.

Thank you so much!

**Comment by danitso-dp** [comment]
Hi @itek09 

I've tried to replicate the `cpu` issue but it changes here. However, it doesn't verify whether the new number of cores and sockets are supported by the hypervisor. I'll have to look into that a bit more to see if I can trigger the problem.

Regarding the disk blocks, this part of the VM resource is still under development, especially when cloning a VM. I'm currently working on adding better support for modifications to these arguments in #7 based on the work in #5.

**Comment by itek09** [comment]
Hi @danitso-dp 

Maybe helps to indicate that I'm not using the root@pam account, I saw that you fixed the default arch issue recently.

I will be following the improvements ;)

**Comment by HarishHary** [comment]
@danitso-dp Is this issue being fixed by the following [PR](https://github.com/danitso/terraform-provider-proxmox/pull/5) ? What's the status on it ?

PS: Great work on the provider ! Best so far !

**Comment by danitso-dp** [comment]
@HarishHary I still don't think the issue is completely fixed but that PR is on my todo. It's hopefully something I can look at again during the next few days as another task required my attention last week.

**Comment by luhahn** [comment]
@HarishHary This PR currently only fixes the disk settings for clones

**Comment by luhahn** [comment]
I also checked on the cpu settings and it works fine for me. I think this issue can be closed, once #5 is merged

**Comment by danitso-dp** [comment]
@itek09 I'm closing this issue as it has been confirmed to work using the latest release. Please don't hesitate to repost this issue, if you still experience problems.
